articles/validating-xml-using-xsd/index.md

```diff
@@ -0,0 +1,262 @@
+status: publish
+published: true
+url: /engineering-education/validating-xml-using-xsd/
+title: Validating XML using XSD
+description: A tutorial on validating XML using XSD
+author: srishilesh-p-s
+date: 2020-11-15T00:00:00-12:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/validating-xml-using-xsd/hero.jpg
+    alt: Validating XML using XSD
+---
+
+In this article, we will understand how validation of XML is done using XSD. By the end of this article, you will get an overview of various techniques for validating XML tags and attributes using XSD. You will also be learning step by step guide for validation. It is highly recommended to go through my [previous article](), to understand the basics of XML and its importance of validation.
+
+<!--more-->
+
+### Table of contents
+- [Introduction](#introduction)
+- [What is XSD?](#what-is-xsd)
+- [Step by step guide for validation](#step-by-step-guide-for-validation)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Introduction
+In our [previous article](), we understood various data serialization techniques like XML, the importance of validating XML, and also validated an XML schema using DTD.
+
+[Document Type Definition (DTD)](https://en.wikipedia.org/wiki/Document_type_definition) helped us in validating the XML schema by parsing the structure of the XML document. But, we can't rely entirely on validation using DTD, since there are high chances of faulty data or data types mismatch in values. This lead to the improvement in the validation technique, where apart from parsing the XML structure, we also understand the semantics of the schema, using XSD.
+
+### What is XSD?
+> According to Wikipedia, [XML Schema Definition (XSD)](https://en.wikipedia.org/wiki/XML_Schema_(W3C)) can be used to express a set of rules to which an XML document must conform to be considered "valid" according to that schema. However, unlike most other schema languages, XSD was also designed with the intent that the determination of a document's validity would produce a collection of information adhering to specific data types.
+
+You may be wondering what is the need for XSD when we already have DTD. XSD can validate much better than DTD, in terms of constraints verification, usage, and relationships between elements and its attributes. According to [W3C](https://www.w3.org/TR/xmlschema11-1/), XSD defines, describes, and catalogs XML vocabularies for classes of XML documents.
+
+### Step by Step guide for validation
+#### XML document
+As you know, XSD is used for complex validation of XML document. So, let's take up some complex XML document for validating. By validating it, understanding of concepts would be much better. In this article, we are not going to cover the basics of building XML document. To keep it simple, let's validate the below XML code. Full XML code can be found [here](https://gist.github.com/srishilesh/5d028a5d9acef3818b014cd4595c9ebe).
```

Comment:
> we are not going to cover the basics of building XML document.

"we are not going to cover the basics of building an XML document." flows better.